### ISSUE TITLE
Redirect or cookie issue with auth mode

### ISSUE THREAD

**Issue by chriske** [thomseddon/traefik-forward-auth#165]
Hi there!

Not so long ago I've tested your application with Traefik v2 in Kubernetes, and it worked like a charm. For my test scenario I used auth mode, OIDC (dex) and a single subdomain like whoami.example.com and auth.example.com for the callback. As I mentioned, everything went well.

Now I'm trying to make work the same setup in a PoC cluster, but with multiple subdomains. For example whoami.k8spoc.example.com and auth.k8spoc.example.com. Now when I hit the whoami domain in the browser, it redirects me to my OIDC providers page and I'm able to log in. In dex's logs I can see that's ok. But after the login was successful, I'm going to be redirected to the OIDC login site again, and not to the whoami webpage. If I check the logs of traefik or the forward-auth container, there is no relevant information at all moreover the forward-auth container logs are completely empty. I tried to set the log level to trace, but there are no logs at all.

Can you help me please figuring out what is happened?

My relevant configs are:

Forward-auth.env:
```
oidc_issuer_url=https://oidc.k8spoc.example.dev
oidc_client_id=oidc
oidc_client_secret=*****
secret=*****
auth_host_url=auth.k8spoc.example.dev
cookie_domain=example.dev
domain=example.hu
```

Middleware:
```
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: traefik-forward-auth
  namespace: ingress
spec:
  forwardAuth:
    address: http://traefik-forward-auth:4181
    authResponseHeaders:
      - X-Forwarded-User
```
Ingress (I'm using certmanager, so I can't make an IngressRoute):
```
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: traefik-forward-auth
  namespace: ingress
  labels:
    app: traefik
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt
    traefik.ingress.kubernetes.io/router.tls: "true"
    traefik.ingress.kubernetes.io/router.entrypoints: websecure
spec:    
  rules:
  - host: auth.k8spoc.example.dev
    http:
      paths:
      - path: /
        backend:
          serviceName: traefik-forward-auth
          servicePort: 4181
  tls:
  - hosts:
    - auth.k8spoc.example.dev
    secretName: auth-https-cert
```
And finally the ingress for whoami app:
```
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: whoami
  labels:
    app: whoami
  namespace: ingress
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt
    traefik.ingress.kubernetes.io/router.tls: "true"
    traefik.ingress.kubernetes.io/router.entrypoints: websecure
    traefik.ingress.kubernetes.io/router.middlewares: ingress-traefik-forward-auth@kubernetescrd
spec:
  rules:
  - host: whoami.k8spoc.example.dev
    http:
      paths:
      - path: /
        backend:
          serviceName: whoami
          servicePort: 80
  tls:
  - hosts:
    - whoami.k8spoc.example.dev
    secretName: whoami-https-cert
```

Thanks in advance!

**Comment by DirtyCajunRice** [comment]
@chriske add
```
traefik.ingress.kubernetes.io/router.middlewares: ingress-traefik-forward-auth@kubernetescrd
```
to your annotations for the oidc provider ingress as well. I know it seems counter-intuitive to auth the auth but this is needed per other issues

**Comment by chriske** [comment]
@DirtyCajunRice thanks for your advice! I did it, and now I get too many redirect error message in the browser.

**Comment by DirtyCajunRice** [comment]
@chriske IIRC I had to look at the 3 way hop happening in dev tools and the tfa config echo'd on container start. ensure all the options/secrets are what you expect

**Comment by chriske** [comment]
@DirtyCajunRice without the extra label on oidc ingress, I can see the communication in dev tools and the redirects. It seems there is no cookie generated, and when the forwarder redirects me back to the original website, it redirects me instantly to oidc site again, because I didn't have the cookie. I've already tried to change the secret multiple times, and the domain configuration too. The strangest thing is that the forward auth container has no logs at all. Not on startup and nothing else. The oidc and auth and whoami sites are on the same domain level. I tried to set it to example.dev and to k8spoc.examleple.dev and it didn't helped.

**Comment by chriske** [comment]
And one more thing: with standard mode (not the auth) this setup is working like a charm with dedicated callback urls. I can see the logs of the container too.

**Comment by DirtyCajunRice** [comment]
If there are no logs in the tfa pod then everything else is moot. With the label on do you see tfa logs?

**Comment by chriske** [comment]
No, nothing. I've tried with TRACE or DEBUG log level, but nothing happened.

**Comment by DirtyCajunRice** [comment]
@chriske add me on discord - there are too many moving parts to this to help you reliably over snail mail like this haha. its `DirtyCajunRice#0001`

**Comment by DirtyCajunRice** [comment]
which also brings up a point... @thomseddon Thoughts on support chat outside github?

**Comment by chriske** [comment]
@DirtyCajunRice thank you for the opportunity! Currently I'm on holiday, I will be back after a week. If it's ok to you too, I will reach you on Discord right after I'm back. Thanks again!

**Comment by thomseddon** [comment]
Hey - if you could bump the log level up to debug and post the logs from the tfa container, that should help point to the issue.

On the support chat - I think that github is probably still the best for most support, because it creates a permanent searchable record for other users to stumble on in the future - but I do agree that it can be a little hard to step through things in realtime! I've created a matrix community (https://matrix.to/#/+traefik-forward-auth:matrix.org) but I probably won't be able to spend much time on there at the moment...

**Comment by chriske** [comment]
@thomseddon @DirtyCajunRice 

Sorry for the late answer guys, unfortunately I had other things to do...
But.. I checked all of my configs, and figured out and solved the issue :)

The problem was with my forward-auth ingress. It must contain the middleware directive to traefik as well, but I didn't added it. 

@DirtyCajunRice maybe this was what you suggested, but you mentioned the oidc ingress, so I interpreted your advice to apply that line to my dex ingress, not to the auth ingress.

Now everything works fine! Thank you for the help and thanks for your work here!
